pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BaseGapfillProcessor.java

```diff
@@ -134,6 +136,9 @@ protected void replaceColumnNameWithAlias(DataSchema dataSchema) {
     for (int i = 0; i < dataSchema.getColumnNames().length; i++) {
       if (columnNameToAliasMap.containsKey(dataSchema.getColumnNames()[i])) {
         dataSchema.getColumnNames()[i] = columnNameToAliasMap.get(dataSchema.getColumnNames()[i]);
+      } else if (columnNameToAliasMap.containsKey(caseInsensitiveTypeString(dataSchema.getColumnNames()[i]))) {
+        dataSchema.getColumnNames()[i] =
+                columnNameToAliasMap.get(caseInsensitiveTypeString(dataSchema.getColumnNames()[i]));
```

Comment:
1. With this code, I think we can remove the first if statement to simplify the code(even though it always runs the `caseInsensitiveTypeString` code and may be a bit slower), what do you think?
```
if (columnNameToAliasMap.containsKey(dataSchema.getColumnNames()[i])) {
    dataSchema.getColumnNames()[i] = columnNameToAliasMap.get(dataSchema.getColumnNames()[i]);
}
```
2. We can define a variable `lowerCaseColumnName = caseInsensitiveTypeString(columnName)` to avoid execute the same code twice.